File: 226. Invert Binary Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root):
        self.rev(root)
        return root
    def rev(self, r):
        if r == None:
            return
        r.right, r.left = r.left, r.right
        self.rev(r.right)
        self.rev(r.left)

File: 226. Invert Binary Tree/Solution.cpp
#include <bits/stdc++.h>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        queue<TreeNode*> q;
        if (!root) return root;
        q.push(root);

        TreeNode *temp, *temp2;

        while (!q.empty()) {
            TreeNode* temp = q.front();
            q.pop();

            temp2 = temp->left;

            temp->left = temp->right;
            temp->right = temp2;
            if (temp->right) q.push(temp->right);
            if (temp->left) q.push(temp->left);
        }

        return root;
    }
};
